junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/InstanceFieldExtensionTest.java

```diff
@@ -8,24 +8,94 @@
 
 import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.MockConsumer;
+import org.apache.kafka.clients.consumer.OffsetResetStrategy;
+import org.apache.kafka.clients.producer.Producer;
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 
+import com.example.StubAnnotation;
+
 import io.kroxylicious.testing.kafka.api.KafkaCluster;
 import io.kroxylicious.testing.kafka.common.BrokerCluster;
 import io.kroxylicious.testing.kafka.invm.InVMKafkaCluster;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.*;
 
 @ExtendWith(KafkaClusterExtension.class)
-public class InstanceFieldExtensionTest extends AbstractExtensionTest {
+class InstanceFieldExtensionTest extends AbstractExtensionTest {
 
     @BrokerCluster(numBrokers = 1)
     KafkaCluster instanceCluster;
 
+    @BrokerCluster(numBrokers = 1)
+    @Name("kafkaCLuster")
+    KafkaCluster namedCluster;
+
+    Consumer<String, String> injectedConsumer;
+
+    Admin injectedAdmin;
+
+    Producer<String, String> injectedProducer;
+
+    @Order(1)
+    Producer<String, String> fieldWithOrderAnnotation;
+
+    final MockConsumer<String, String> mockConsumer = new MockConsumer(OffsetResetStrategy.LATEST);
+
+    @StubAnnotation
+    Admin fieldWithUnrecognizedAnnotation;
+
+    @SuppressWarnings("DeprecatedIsStillUsed")
+    @Deprecated
+    Admin fieldWithJavaLangAnnotations;
+
+    @Name("kafkaCLuster")
+    Admin namedAdmin;
+
+    @Test
+    void shouldInjectConsumerField() {
+        assertThat(injectedConsumer).isNotNull().isInstanceOf(Consumer.class);
+    }
+
+    @Test
+    void shouldInjectProducerField() {
+        assertThat(injectedProducer).isNotNull().isInstanceOf(Producer.class);
+    }
+
+    @Test
+    void shouldInjectAdminField() {
+        assertThat(injectedAdmin).isNotNull().isInstanceOf(Admin.class);
+    }
+
+    @Test
+    void shouldNotInjectIntoInitialisedField() {
+        assertThat(mockConsumer).isNotNull().isInstanceOf(MockConsumer.class);
+    }
+
+    @Test
+    void shouldInjectIntoFieldsWithRecognisedAnnotation() {
+        assertThat(fieldWithOrderAnnotation).isNotNull().isInstanceOf(Producer.class);
+        assertThat(fieldWithJavaLangAnnotations).isNotNull().isInstanceOf(Admin.class);
+        assertThat(namedAdmin).isNotNull().isInstanceOf(Admin.class);
+    }
+
+    @Test
+    void shouldInjectIntoFieldWithJunitAnnotation() {
+        assertThat(fieldWithOrderAnnotation).isNotNull().isInstanceOf(Producer.class);
+        // TODO should we use the producer to send a record?
+    }
+
+    @Test
+    void fieldWithUnrecognizedAnnotationsNotInjected() {
+        assertNull(fieldWithUnrecognizedAnnotation);
```

Comment:
You've. mostly adopted assertThat.. might as well make the remaining tests use the same.
```suggestion
        assertThat(fieldWithUnrecognizedAnnotation).isNull();
```